api/nodes/views.py

```diff
@@ -956,7 +956,21 @@ def get_default_queryset(self):
 
     # overrides ListAPIView
     def get_queryset(self):
-        return self.get_queryset_from_request().distinct()
+        path = self.kwargs[self.path_lookup_url_kwarg]
+        # query param info when used on a folder gives that folder's metadata instead of the metadata of it's children
+        if 'info' in self.request.query_params and path.endswith('/'):
```

Comment:
It might be more user-friendly to raise an error if `info` is passed on a non-folder.

Reply:
I don't really think so. Imagine a someone wants metadata on a path and they don't know if it's a file or a folder, with a 'info' parameter they are guaranteed that that item's metadata. If it throws an error they'll have to put in logic to retry the request without the parameter.

Reply:
Ah, I didn't realize that the client receives the same metadata for files as folders with `?info=true`. In that case, I agree with you, and this is fine as is.